File: treino-abc-backend/src/main/java/com/treino_abc_backend/entity/Exercicio.java
package com.treino_abc_backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@Entity
@Getter
@Setter
@Table(name = "treino_exercicio")
public class Exercicio {

    @Id
    @GeneratedValue
    @Column(columnDefinition = "BINARY(16)")
    private UUID id;

    @Column(nullable = false)
    private String nome;

    @Column(nullable = false)
    private String grupoMuscular;

    @Column(nullable = false)
    private int series;

    @Column(nullable = false)
    private int repMin;

    @Column(nullable = false)
    private int repMax;

    @Column(nullable = false)
    private int pesoInicial;

    @Column(length = 500)
    private String observacao;
}

File: flutter_application_treinoabc/lib/models/exercicio.dart
class Exercicio {
  final String id;
  final String nome;
  final String grupoMuscular;
  final int series;
  final int repMin;
  final int repMax;
  final double pesoInicial;
  final String? observacao;

  Exercicio({
    required this.id,
    required this.nome,
    required this.grupoMuscular,
    required this.series,
    required this.repMin,
    required this.repMax,
    required this.pesoInicial,
    this.observacao,
  });

  factory Exercicio.fromJson(Map<String, dynamic> json) {
    return Exercicio(
      id: json['id'] ?? '',
      nome: json['nome'] ?? '',
      grupoMuscular: json['grupoMuscular'] ?? '',
      series: json['series'] ?? 0,
      repMin: json['repMin'] ?? 0,
      repMax: json['repMax'] ?? 0,
      pesoInicial: (json['pesoInicial'] as num?)?.toDouble() ?? 0.0,
      observacao: json['observacao'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nome': nome,
      'grupoMuscular': grupoMuscular,
      'series': series,
      'repMin': repMin,
      'repMax': repMax,
      'pesoInicial': pesoInicial,
      'observacao': observacao,
    };
  }
}
